docs/state/substates.md

```diff
@@ -5,63 +5,109 @@ from pcweb.templates.docpage import docdemo_from
 ```
 
 # Substates
-As your app grows, your state will grow too.  You can split your state into
-multiple substates from your base state to keep things organized.
+A substate is a class that inherits from another state class, providing a means for two state classes to share common state resources, such as variables or event handlers.
 
 ## Creating a Substate
-Your base state should inherit from `rx.State`. Substates can either inherit
-from your base state or other substates.
+A substate can typically inherit from a base state(a state that inherits from `rx.State`) or another substate.
+Let’s take a look at a simple example illustrating how and when to use substates. 
+The code below shows two states, `RectangleState` and `SquareState`, each representing metadata about rectangles and squares, along with the calculation of their respective areas.
 
 ```python exec
-class ParentState(State):
-    checked: bool = True
-    count: int = 0
+class RectangleState(rx.State):
```

Comment:
I think a better example to use here would be something like `HomePageState` and `LoginPageState`. The way it is currently is similar to Python classes/inheritance but the purpose of substates is slightly different, it's more to provide separation of states rather than really doing anything with inheritance.

Basically the way it's structured now evokes the idea that substates need to be similar in terms of what vars they have, which may cause confusion.